File: JavaIntroNeuralNetworkEdition2/src/com/heatonresearch/httprecipes/spider/workload/sql/oracle/OracleWorkloadManager.java
package com.heatonresearch.httprecipes.spider.workload.sql.oracle;

import java.sql.*;

import com.heatonresearch.httprecipes.spider.workload.sql.*;

/**
 * The Heaton Research Spider 
 * Copyright 2007 by Heaton Research, Inc.
 * 
 * HTTP Programming Recipes for Java ISBN: 0-9773206-6-9
 * http://www.heatonresearch.com/articles/series/16/
 * 
 * OracleWorkloadManager: This workload manager is compatible
 * with the Oracle database.
 * 
 * This class is released under the:
 * GNU Lesser General Public License (LGPL)
 * http://www.gnu.org/copyleft/lesser.html
 * 
 * @author Jeff Heaton
 * @version 1.1
 */
public class OracleWorkloadManager extends SQLWorkloadManager
{
  public SQLHolder createSQLHolder()
  {
    return new OracleHolder();
  }

  /**
   * Return the size of the specified column.
   *
   * @param table
   *          The table that contains the column.
   * @param column
   *          The column to get the size for.
   * @return The size of the column.
   * @throws SQLException
   *           For SQL errors.
   */
  public int getColumnSize(String table, String column) throws SQLException
  {
    ResultSet rs = this.getConnection().getMetaData().getColumns(null, null,
        table, "%");
    while (rs.next())
    {

      String c = rs.getString("COLUMN_NAME");
      int size = rs.getInt("COLUMN_SIZE");
      if (c.equalsIgnoreCase(column))
      {
        return size;
      }
    }
    return -1;
  }
}

File: CSharpHTTPRecipesEdition1/HTTPRecipes/HeatonResearchSpider/Spider/Workload/SQL/Oracle/OracleWorkloadManager.cs
// The Heaton Research Spider for .Net 
// Copyright 2007 by Heaton Research, Inc.
// 
// From the book:
// 
// HTTP Recipes for C# Bots, ISBN: 0-9773206-7-7
// http://www.heatonresearch.com/articles/series/20/
// 
// This class is released under the:
// GNU Lesser General Public License (LGPL)
// http://www.gnu.org/copyleft/lesser.html
//
using System;
using System.Collections.Generic;
using System.Text;
using System.Data.Common;
using System.Data;

namespace HeatonResearch.Spider.Workload.SQL.Oracle
{
    /// <summary>
    /// OracleWorkloadManager: Contains a workload manager 
    /// customized to Oracle.
    /// </summary>
    class OracleWorkloadManager : SQLWorkloadManager
    {
        /// <summary>
        /// Creates an instance of the OracleHolder class.
        /// </summary>
        /// <returns>An SQLHolder derived object.</returns>
        public override SQLHolder CreateSQLHolder()
        {
            return new OracleHolder();
        }


        /// <summary>
        /// Return the size of the specified column.  Oracle
        /// requires the "%" parameter to be specified.
        /// </summary>
        /// <param name="table">The table that contains the column.</param>
        /// <param name="column">The column to get the size for.</param>
        /// <returns>The size of the column.</returns>
        public override int GetColumnSize(String table, String column)
        {
            String[] restriction = { null, null, table, "%" };
            DataTable dt = this.Connection.GetSchema("Columns", restriction);
            foreach (System.Data.DataRow row in dt.Rows)
            {

                if (String.Compare(row["column_name"].ToString(), column, true) == 0)
                {
                    return int.Parse(row["column_size"].ToString());
                }
            }

            return -1;
        }
    }
}
